### ISSUE TITLE
Child Modules Code Example

### ISSUE THREAD

**Issue by Lucas-Steinmann** [PyTorchLightning/pytorch-lightning#3124]
## 📚 Documentation

For typos and doc fixes, please go ahead and:

1. Create an issue.
My issues are all about the documentation page about child modules: https://pytorch-lightning.readthedocs.io/en/latest/child_modules.html#child-modules

a) The code example refers to a variable named ``logits`` in the method ``_shared_eval(..)``.
What is logits referring to?

b) ``forward`` returns the result of calling ``decoder``. This is correctly used in ``training_step``. But ``_shared_eval`` seems to expect a classifier output when calling ``self(representation)``.

c) Even more confusing in the third code block, calling the autoencoder returns the representation:
```
some_images = torch.Tensor(32, 1, 28, 28)
representations = autoencoder(some_images)
```

So in 
1. training_step -> decoding
2. _shared_eval -> classifier
3. inference -> encoding

Or is there some underlying logic which switches between the implementations?

d) Probably connected with the other issues: The ``forward``-function of ``LitMNIST``, which implements the classficiation, required in ``_shared_eval``, is overriden, but not accessed.

2. Fix the typo.   
No typo.

Not sure what the correct way to rewrite the example is. 
I can fix the issues if someone answers my questions/explains the issues I presented in 1.

3. Submit a PR.
Will do, when I know how to fix the issues.

**Comment by Borda** [comment]
mind have look @rohitgr7 🐰

**Comment by rohitgr7** [comment]
yeah it's a mistake? Mind send a PR @Lucas-Steinmann :)  ??

**Comment by Lucas-Steinmann** [comment]
Since I didn't get a definite answer, whether the intention is to implement a Classifier or only an AutoEncoder, I went with my intuition and removed the classifier parts.

This is a small change, which solves only part of the problem.
I think the following problems should be solved, to further improve ease of understanding of the Child Modules section:

1. It is unclear which LitMNIST module the code refers to. This is exaggerated by 2.
2. The Child Modules is a separate section, accessible through the menu __and__ a subsection of the step-by-step walkthrough. If someone accesses it through the menu, he might have never seen the LitMNIST module.
3. Is it that good of a practise to extend a classification module and use it for something completely different? The weights of the classificator would still be present and the semantics of the forward function changes drastically. (think from an OO-perspective)

My suggestions are:

1. Use a lightning data module instead of overriding
2. 2.1 Remove the Child Modules section from the step-by-step walkthrough. That walkthrough is already too large anyway, __or__ 
 2.2 Rewrite the section or the previous sections of the guide, such that they really work in a step-by-step fashion and link to used modules explicetly, such that people coming from the menu understand what other parts of code are requiered.

I'm happy to apply those suggestions. But I would like to get approval of that direction in beforehand. We can also discuss them in a new issue.

**Comment by rohitgr7** [comment]
@Lucas-Steinmann I'll check this later and comment on the PR itself.
